fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/client/BulkDataClient.java

```diff
@@ -312,11 +309,139 @@ public PollingLocationResponse status(String job) throws Exception {
 
     /**
      * @param job
-     * @return
+     * @return status code
      * @throws Exception
      */
-    public void delete(String job) throws Exception {
-        // Example: https://localhost:9443/ibm/api/batch/jobexecutions/9
+    public Response.Status delete(String job) throws Exception {
+        // <b>Code Flow</b>
+        // 1 - Check if the Job Exists and is Valid
+        // 2 - Check if the Tenant is expected
+        // 3 - Try to immediately DELETE (Optimistic)
+        //      A - SC_NO_CONTENT = SUCCESS GO_TO End
+        //      B - !SC_INTERNAL_ERROR GO_TO STOP 
+        //      C - ERROR GO_TO Error 
+        // 4 - Try to stop the PUT /ibm/api/batch/jobinstances/<job>?action=stop
+        //     A - SC_CONTENT_REDIRECT (302) GO_TO Location to PUT
+        //       - Stop requests sent to the batch REST API must be sent directly to the executor where the job is running.
+        //       - Link - https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_batch_rest_api.html#rwlp_batch_rest_api__STOP_requests
+        //     B - SC_ACCEPTED (202) GO_TO ACCEPTED
+        // RETURN - ACCEPTED: SC_ACCEPTED - STILL PROCESSING
+        // RETURN - DONE: SC_OK - DONE DELETING
+
+        // 1 and 2 (We know it exists)
+        runVerificationOfTenantForJob(job);
+
+        // 3 optimistic delete (empty)
+        Response.Status status = runDeleteOfTenantForJob(job);
+
+        // Check 3 (B), else fall through (A).
+        // Default Response is Accepted (202)
+        // The Alternative Response is 200
+
+        // Check for a server-side error
+        if (HttpStatus.SC_INTERNAL_SERVER_ERROR == status.getStatusCode()) {
+            // 3.C - ERROR Condition
+            // The Server hit an error 
+            throw BulkDataExportUtil.buildOperationException(
+                    "Deleting the job has failed; the content is not abandonded", IssueType.EXCEPTION);
+        } else if (HttpStatus.SC_NO_CONTENT != status.getStatusCode()) {
+            // The Delete is unsuccessful (SC_NO_CONTENT)
+            // 3.B - STOP Condition and now step 4 in the flow.
+            status = runStopOfTenantForJob(job);
+        }
+        return status;
+    }
+
+    /*
+     * stops the job (local or remote batch processor).
+     */
+    private Response.Status runStopOfTenantForJob(String job) throws Exception {
```

Comment:
the function name sounds a little bit strange ...  maybe runStopJobForTenant() is better...

Reply:
changed